financial-services-accelerator/components/org.wso2.financial.services.accelerator.gateway/src/main/java/org/wso2/financial/services/accelerator/gateway/executor/impl/dcr/DCRExecutor.java

```diff
@@ -0,0 +1,230 @@
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+import javax.ws.rs.HttpMethod;
+
+/**
+ * Executor for DCR.
+ */
+public class DCRExecutor implements FinancialServicesGatewayExecutor {
+
+    private static final Log log = LogFactory.getLog(DCRExecutor.class);
+    private static final Map<String, Object> configs = GatewayDataHolder.getInstance()
+            .getFinancialServicesConfigurationService().getConfigurations();
+
+    @Override
+    public void preProcessRequest(FSAPIRequestContext fsapiRequestContext) {
+
+        if (fsapiRequestContext.isError()) {
+            return;
+        }
+
+        boolean validateJWT = true;
+        if (configs.containsKey(FinancialServicesConstants.VALIDATE_JWT)) {
+            validateJWT = Boolean.parseBoolean(configs.get(FinancialServicesConstants.VALIDATE_JWT).toString());
+        }
+
+        if (validateJWT) {
+            String payload = fsapiRequestContext.getRequestPayload();
+            try {
+                String httpMethod = fsapiRequestContext.getMsgInfo().getHttpMethod();
+                if (HttpMethod.POST.equals(httpMethod) || HttpMethod.PUT.equals(httpMethod)) {
+                    if (payload != null) {
+                        //decode request jwt
+                        String decodedRequest = JWTUtils.decodeRequestJWT(payload, "body");
+
+                        //Check whether decodedRequest is null
+                        if (decodedRequest == null) {
+                            handleBadRequestError(fsapiRequestContext,
+                                    "Provided jwt is malformed and cannot be decoded");
+                            return;
+                        }
+
+                        JSONObject decodedRequestObj = new JSONObject(decodedRequest);
+                        JSONObject decodedSSA;
+                        //Check whether the SSA exists and decode the SSA
+                        if (decodedRequestObj.has(GatewayConstants.SOFTWARE_STATEMENT) &&
+                                decodedRequestObj.getString(GatewayConstants.SOFTWARE_STATEMENT) != null) {
+                            String ssa = JWTUtils.decodeRequestJWT(decodedRequestObj
+                                    .getString(GatewayConstants.SOFTWARE_STATEMENT), "body");
+                            decodedSSA = new JSONObject(ssa);
+                        } else {
+                            handleBadRequestError(fsapiRequestContext,
+                                    "Required parameter software statement cannot be null");
+                            return;
+                        }
+                        // Validate the request signature
+                        JWTClaimsSet requestClaims = GatewayUtils.validateRequestSignature(payload, decodedSSA);
+                        // Construct the IS DCR request payload
+                        String isDcrPayload = GatewayUtils.constructIsDcrRequestPayload(requestClaims, decodedSSA);
```

Comment:
Shall we rename this as dcrISPayload (similar to usage in GatewayUtils)? This might be confused as a boolean.

Reply:
Fixed in https://github.com/wso2/financial-services-accelerator/pull/229/commits/9aa8af74e08881a013ed94cc0fcf30547bcf0e09